.github/actions/gradle-build/action.yml

```diff
@@ -0,0 +1,35 @@
+#TODO move this action to https://github.com/micronaut-projects/github-actions
+name: Build steps for GitHub gradle workflow
+description: Perform build steps for gradle workflow
+inputs:
+  java:
+    description: java version
+    required: true
+    type: string
+  gradle-enterprise-access-key:
+    description: gradle enterprise access key
+    required: true
+    type: string
+  gradle-enterprise-cache-username:
+    description: gradle enterprise cache username
+    required: true
+    type: string
+  gradle-enterprise-cache-password:
+    description: gradle enterprise cache password
+    required: true
+    type: string
+runs:
+  using: "composite"
+  steps:
+    - name: Build with Gradle
+      shell: bash
+      run: |
+        # Awful hack for kapt and JDK 16. See https://youtrack.jetbrains.com/issue/KT-45545
+        if [ ${{ inputs.java }} == 16 ]; then export GRADLE_OPTS="-Dorg.gradle.jvmargs=--illegal-access=permit"; fi
```

Comment:
This can probably be removed now, the linked bug is fixed

Reply:
removed this fix